compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -3177,7 +3177,7 @@ object Parsers {
     def pattern1(location: Location = Location.InPattern): Tree =
       val p = pattern2(location)
       if in.isColon then
-        val isVariableOrNumber = isVarPattern(p) || p.isInstanceOf[Number]
+        val isVariableOrNumber = isVarPattern(p) || p.isInstanceOf[Number] || isBackquoted(p)
```

Comment:
Scala 2 honors the requirement even in backquoting.
```
scala> 42 match { case `Int`: Int => `Int` }
                            ^
       error: Pattern variables must start with a lower-case letter. (SLS 8.1.1.)
```

Reply:
Normal syntax for non-var: `42 match { case X @ (_: Int) => X }`

Maybe it's worth adding a neg test.